### PR TITLE
add periodic 2.1 job

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

**Review by stefannica — COMMENTED** [review]
+1 from me, but I'll leave it for @flaviodsr to approve

File: .github/workflows/periodic-21.yml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        run: |
+          kubectl delete ns ${NAMESPACE} || true
+          kubectl create ns ${NAMESPACE}
+      - name: Get ingress IP
+        run: |
+          echo "::set-env name=INGRESS_IP::$(kubectl get svc nginx-ingress-controller -n nginx-ingress -o json | jq -r ".status.loadBalancer.ingress[].ip")"
+      - name: Generate helm values file for CI
+        run: |
+          cat << EOF | tee ${NAMESPACE}.yaml
+          expose:
```
Comment (flaviodsr) [inline .github/workflows/periodic-21.yml:67]:
```suggestion
          cat << EOF | tee ${NAMESPACE}.yaml
          suse:
            AcceptBetaEULA: "yes"
          expose:
```

File: .github/workflows/periodic-21.yml [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+          env: ${{ steps.setup.outputs.env_name }} (${{ matrix.source }})
+          ref: ${{ steps.setup.outputs.env_ref }}
+      - name: SUSE Private Registry (install)
+        env:
+          HELM_EXPERIMENTAL_OCI: 1
+        run: |
+          helm chart pull ${{ matrix.install_src }}/charts/registry/harbor:latest
+          helm chart export ${{ matrix.install_src }}/charts/registry/harbor:latest
+          chart="./harbor"
+          # replace images repository according to source
```
Comment (flaviodsr) [inline .github/workflows/periodic-21.yml:93]:
```suggestion
        run: |
          chart="./harbor"
          rm -rf ${chart}
          helm chart pull ${{ matrix.install_src }}/charts/registry/harbor:latest
          helm chart export ${{ matrix.install_src }}/charts/registry/harbor:latest
          # replace images repository according to source
```

File: .github/workflows/periodic-21.yml [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+          helm install ${NAMESPACE} ${chart} -n ${NAMESPACE} --values ${NAMESPACE}.yaml --timeout 8m --wait
+      - name: SUSE Private Registry (upgrade)
+        if: matrix.upgrade_src
+        env:
+          HELM_EXPERIMENTAL_OCI: 1
+        run: |
+          helm chart pull ${{ matrix.upgrade_src }}/charts/registry/harbor:latest
+          helm chart export ${{ matrix.upgrade_src }}/charts/registry/harbor:latest
+          chart="./harbor"
+          # replace images repository according to source
```
Comment (flaviodsr) [inline .github/workflows/periodic-21.yml:104]:
```suggestion
        run: |
          chart="./harbor"
          rm -rf ${chart}
          helm chart pull ${{ matrix.upgrade_src }}/charts/registry/harbor:latest
          helm chart export ${{ matrix.upgrade_src }}/charts/registry/harbor:latest
          # replace images repository according to source
```

File: .github/workflows/periodic-21.yml [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            upgrade_src: registry.suse.de/suse/sle-15-sp2/update/products/caps-registry/2.1
+    steps:
+      - name: Setup environment for job triggered by ${{ github.event_name }}
+        id: setup
+        run: |
+          case "${GITHUB_EVENT_NAME}" in
+            schedule|workflow_dispatch)
+              echo "::set-output name=env_ref::release-2.1"
+              echo "::set-output name=env_name::periodic-2.1"
+              echo "::set-env name=NAMESPACE::spr20-sched-${{ matrix.source }}"
```
Comment (flaviodsr) [inline .github/workflows/periodic-21.yml:39]:
```suggestion
              echo "::set-env name=NAMESPACE::spr21-sched-${{ matrix.source }}"
```

File: .github/workflows/periodic-21.yml (hunk 2 above)
Comment (flaviodsr) [reply .github/workflows/periodic-21.yml:93]: This is required to remove the chart pulled by the previous run

File: .github/workflows/periodic-21.yml (hunk 3 above)
Comment (flaviodsr) [reply .github/workflows/periodic-21.yml:104]: same here

**Review by flaviodsr — CHANGES_REQUESTED** [review]
I have tested this on my fork and with the suggested changes the workflow should be fine.

**Review by flaviodsr — APPROVED** [review]
LGTM

**Pull request merged**